### PR TITLE
Add retry mechanism for 429 Too Many Requests and 500 Internal Server Error

### PR DESCRIPTION
**Pull request by dmytrokoren**

### CODE REVIEW

**Comment by dmytrokoren** [comment]
@jdholtz lets review and merge.

File: lib/reservation_monitor.py [hunk 1]
```diff
@@ -194,42 +197,63 @@ def _check(self) -> bool:
...
+                    else:
+                        logger.debug(
+                            "Error (status: %d) persists. Skipping reservation retrieval",
+                            err.status_code,
+                        )
+                        self.notification_handler.too_many_requests_during_login()
+                else:
+                    logger.debug("Error logging in. %s. Exiting", err)
+                    self.notification_handler.failed_login(err)
+                    time.sleep(1)
```
Comment (jdholtz) [inline lib/reservation_monitor.py:253]: Is this sleep call needed? The notification has still been sent successfully if I remove the call (I believe it is all done synchronously). It's a very small nitpick though, so I can just remove it later if we decide that it isn't needed.

**Review by jdholtz — APPROVED** [review]
Thanks for separating out this change to a separate PR. I updated the tests to mock the sleep calls (so the unit tests don't take 20+ seconds) and it will be a part of next release.

File: lib/reservation_monitor.py (hunk 1 above)
Comment (dmytrokoren) [reply lib/reservation_monitor.py:253]: We can probably remove it, I don't remember why it's there )

**Pull request merged**